lib/PlatformParser.php

```diff
@@ -0,0 +1,168 @@
+<?php
+namespace LMMS;
+use LMMS\Platform;
+
+enum Architecture: string {
+    case Intel = "32-bit";
+    case Intel64 = "Intel";
+    case Arm = "ARM";
+    case Arm64 = "ARM64";
+    case Riscv = "RISC-V 32-bit";
+    case Riscv64 = "RISC-V";
+    case Ppc = "PowerPC 32-bit";
+    case Ppc64 = "PowerPC";
+    case Unknown = "Unknown";
+}
+
+enum Qualifier: string {
+	case Mingw = "mingw";
+	case Msvc = "msvc";
+	case Gcc = "gcc";
+	case Clang = "clang";
+	case Vcpkg = "vcpkg";
+	case Msys2 = "msys2";
+	case Unknown = "Unknown";
+}
+
+class PlatformParser {
+	const ARCHITECTURE_PATTERNS = [
+		// 64 first: matches x86_64 before x86, etc
+		Architecture::Intel64->value => ['x86_64', 'amd64', 'mingw64', 'win64'],
+		Architecture::Intel->value => ['x86', 'i386', 'i686', 'mingw32', 'win32'],
+		Architecture::Arm64->value => ['arm64', 'aarch64', 'armv8'],
+		Architecture::Arm->value => ['arm', 'arm32', 'armv7'],
+		Architecture::Riscv64->value => ['riscv64'],
+		Architecture::Riscv->value => ['riscv'],
+		Architecture::Ppc64->value => ['ppc64', 'powerpc64'],
+		Architecture::Ppc->value => ['ppc', 'powerpc'],
+	];
+
+	const PLATFORM_PATTERNS = [
+		Platform::Linux->value => ['linux', '.AppImage', '.run', '.rpm', '.deb'],
+		Platform::Windows->value => ['windows', 'win32', 'win64', 'msys2', 'mingw32', 'mingw64', '.exe', 'msvc'],
+		Platform::MacOS->value => ['mac', 'osx', '.pkg', '.dmg'],
+	];
+
+	const QUALIFIER_PATTERNS = [
+		Qualifier::Mingw->value => ['mingw'],
+		Qualifier::Msvc->value => ['msvc'],
+		Qualifier::Clang->value => ['clang', 'llvm'],
+		Qualifier::Vcpkg->value => ['vcpkg'],
+		Qualifier::Msys2->value => ['msys2'],
+		Qualifier::Gcc->value => ['gcc'],
```

Comment:
I just realized that there's a small chance that this `gcc` value could match on commit hash...  🤔

Reply:
My hex brain wasn't working when I wrote that.... it stops at `f`, so we're good. 😅